File: BackEnd/gamedex/src/main/java/cat/copernic/gamedex/entity/Category.java
package cat.copernic.gamedex.entity;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Classe que representa una categoria de videojocs.
 */
@Document(collection = "category")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Category {

    /**
     * El nom de la categoria.
     */
    @Id
    private String nameCategory;

    /**
     * La descripció de la categoria.
     */
    private String description;

    /**
     * La foto de la categoria en format byte array.
     */
    @Field(name = "category_photo")
    private byte[] categoryPhoto;
}

File: FrontEnd/app/src/main/java/cat/copernic/grup4/gamedex/Core/Model/Category.kt
package cat.copernic.grup4.gamedex.Core.Model

/**
 * Classe que representa una categoria de videojocs.
 *
 * @param nameCategory El nom de la categoria.
 * @param description La descripció de la categoria.
 * @param categoryPhoto La foto de la categoria en format Base64 (opcional).
 */
data class Category(
    val nameCategory: String,
    val description: String,
    val categoryPhoto: String? = null
)